proxy/server/target.go

```diff
@@ -141,8 +164,39 @@ func (s *TargetStream) Send(req proto.Message) error {
 // All data received from target will be converted into ProxyReply
 // messages for sending to a proxy client, including the final
 // status of the target stream
-func (s *TargetStream) Run(replyChan chan *pb.ProxyReply) {
-	group, ctx := errgroup.WithContext(s.grpcStream.Context())
+func (s *TargetStream) Run(nonce uint32, replyChan chan *pb.ProxyReply) {
+	startStreamReply := &pb.ProxyReply{
+		Reply: &pb.ProxyReply_StartStreamReply{
+			StartStreamReply: &pb.StartStreamReply{
+				Target: s.target,
+				Nonce:  nonce,
+			},
+		},
+	}
+
+	grpcStream, err := s.grpcConn.NewStream(s.ctx, s.serviceMethod.StreamDesc(), s.serviceMethod.FullName())
```

Comment:
you aren't locking here. Should you be?

Reply:
no. Most of this method uses the local-in-scope stream object (after associating it with the target stream to make it visible to other callers.)

_Technically_ speaking, the lock is currently unnecessary, because I don't pass back the stream ID to the caller until after the connected stream has been associated with the target, but .. putting in the mutex helps with any future cases in which we might try to access the stream prior to 'Run', and I didn't feel right about depending on the organization of the current code for threadsafety.